waiter/src/waiter/main.clj

```diff
@@ -117,6 +118,7 @@
     (reify Thread$UncaughtExceptionHandler
       (uncaughtException [_ thread throwable]
         (log/error throwable (str (.getName thread) " threw exception: " (.getMessage throwable))))))
+  (jvm-metrics/instrument-jvm)
```

Comment:
And based my understanding, the only time we ask for the value of the jvm gauges is in the handler.  We do not evaluate them automatically on a regular basis.